ecommerce/views/v0/__init__.py

```diff
@@ -425,110 +423,76 @@ def post(self, request, *args, **kwargs):
-            # Transaction held for review in the payment processor's system
-            # The transaction is in limbo here - it may be approved or denied
-            # at a later time
-            self.logger.debug(
-                "Transaction flagged for review: {msg}".format(
-                    msg=processor_response.message
-                )
+                reverse("cart"), {"type": USER_MSG_TYPE_PAYMENT_DECLINED}
             )
+        elif order_state == Order.STATE.REVIEW:
+            basket = Basket.objects.filter(user=order.purchaser).first()
             if basket:
                 if basket.has_user_blocked_products(order.purchaser):
                     return redirect_with_user_message(
                         reverse("user-dashboard"),
                         {"type": USER_MSG_TYPE_ENROLL_BLOCKED},
                     )
                 else:
-                    basket.delete()
-
-            order.review(payment_data)
-            order.save()
-
-            if basket and basket.compare_to_order(order):
-                basket.delete()
-
+                    return redirect_with_user_message(
+                        reverse("user-dashboard"),
+                        {"type": USER_MSG_TYPE_PAYMENT_REVIEW},
+                    )
+        elif order_state == Order.STATE.FULFILLED:
             return redirect_with_user_message(
-                reverse("user-dashboard"), {"type": USER_MSG_TYPE_PAYMENT_REVIEW}
+                reverse("user-dashboard"),
+                {
+                    "type": USER_MSG_TYPE_PAYMENT_ACCEPTED,
+                    "run": order.lines.first().purchased_object.course.title,
+                },
             )
-        elif processor_response.state == ProcessorResponse.STATE_ACCEPTED:
-            # It actually worked here
-            self.logger.debug(
-                "Transaction accepted!: {msg}".format(msg=processor_response.message)
-            )
-
-            return api.fulfill_completed_order(order, payment_data, basket)
         else:
-            order.cancel()
-            order.save()
             return redirect_with_user_message(
                 reverse("user-dashboard"), {"type": USER_MSG_TYPE_PAYMENT_ERROR_UNKNOWN}
             )
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+class BackofficeCallbackView(APIView):
+    authentication_classes = []  # disables authentication
+    permission_classes = []  # disables permission
+
+    def post(self, request, *args, **kwargs):
+        """
+        This endpoint is called by Cybersource as a server-to-server call
+        in order to respond with the payment details.
+        """
+        order = api.get_order_from_cybersource_payment_response(request)
+
+        # We only want to process responses related to orders which are PENDING
+        # otherwise we can conclude that we already received a resonse through
+        # the user's browser.
+        if order is not None and order.state == Order.STATE.PENDING:
```

Comment:
If we can't find an order in this case, we should probably log an error regarding it.

Reply:
@rhysyngsun should this still respond with a `200` if there is no order found and an exception is raised, or a `404`?